src/erlazure_insights.erl

```diff
@@ -0,0 +1,140 @@
+get_events_all(SubscriptionId, Filter, Select, Config=#azure_config{}) ->
+    {ok, Values, NextToken} = get_events(SubscriptionId, Filter, Select, Config, []),
+    case NextToken of
+        undefined -> {ok, Values};
+        _ -> follow_token(binary_to_list(NextToken), Values, Config)
+    end.
+
+-spec get_events(string(), string(), string(), #azure_config{}, string()) ->
+    {ok, list(), undefined | binary()}.
+get_events(SubscriptionId, Filter, Select, Config=#azure_config{}, SkipToken) ->
+    case SkipToken of
+        [] ->
+            case http_request(SubscriptionId, Filter, Select, Config) of
+                {ok, _Values, _NextToken} = Response ->
+                    Response;
+                {error, _Reason} = ErrorCase ->
+                    ErrorCase
+            end;
+        _ -> follow_token(SkipToken, [], Config)
+    end.
+
+-spec follow_token(string(), list(), #azure_config{}) ->
+    {ok, list()} | {{error, term()}, {accumulated, list()}}.
+follow_token(Url, DataAcc, Config) ->
+    Result = request_api(Url, Config),
+    case Result of
+        {ok, Data, undefined} ->
+            {ok, Data};
+        {ok, Data, NextToken} ->
+            follow_token(binary_to_list(NextToken), [DataAcc | Data], Config);
+        {error, _Reason} = ErrorCase ->
+            {ErrorCase, {accumulated, DataAcc}}
+    end.
+
+request_api(Url, Config) ->
+    Response = httpc:request(get, {Url, construct_headers(Config)}, [], []),
+
+    case Response of
+        {ok, {{_Version, 200, _Phrase}, _Headers, Body}} ->
+            DecodedBody = jsx:decode(list_to_binary(Body)),
+            NextLink = proplists:get_value(<<"nextLink">>, DecodedBody, undefined),
+            Values = proplists:get_value(<<"value">>, DecodedBody, undefined),
+            {ok, Values, NextLink};
+        {ok, {{_Version, Code, Phrase}, _Headers, Body}} ->
+            {error, {Code, {Phrase, Body}}};
+        {error, _Reason} = ErrorCase ->
+            ErrorCase
+    end.
+
+http_request(SubscriptionId, Filter, Select, Config) ->
+    URL = "https://management.azure.com/subscriptions/" ++
+    SubscriptionId ++
+    "/providers/microsoft.insights/eventtypes/management/values?" ++ "api-version=2014-04-01" ++ 
+    "&%24filter=" ++ Filter ++
+    "&%24select=" ++ Select,
+    request_api(URL, Config).
+
+-spec get_time_offset_as_timestamp(datetime(), offset()) -> string().
+get_time_offset_as_timestamp(DateTime = {_Date, _Time}, {Unit, Offset}) ->
+
+    Seconds = calendar:datetime_to_gregorian_seconds(DateTime),
+
+    NewDateTime = case Unit of
+        seconds -> calendar:gregorian_seconds_to_datetime(Seconds - Offset);
+        minutes -> calendar:gregorian_seconds_to_datetime(Seconds - Offset * 60);
+        hours   -> calendar:gregorian_seconds_to_datetime(Seconds - Offset * 60 * 60)
+    end,
+
+    erlazure_utils:iso_8601_fmt(NewDateTime).
+
+-spec create_time_bounded_filter(string(), string()) -> string().
+create_time_bounded_filter(TS1, TS2) ->
```

Comment:
Since this function is supposed to accept eventTimestamp in form of "2014-12-29T22:00:37Z", 

should we also add new function to `azure_utils` module, which will convert Timestamp to `iso_8601` format?